core/src/utils/directive.ts

```diff
@@ -235,7 +235,12 @@ export interface AttributesDirectiveProps {
 	 * @remarks
 	 * Key-value pairs where keys are event types and values are event handlers.
 	 */
-	events?: Partial<{[K in keyof HTMLElementEventMap]: (this: HTMLElement, event: HTMLElementEventMap[K]) => void}>;
+	events?: Partial<{
+		[K in keyof HTMLElementEventMap]: {
+			eventFn: (this: HTMLElement, event: HTMLElementEventMap[K]) => void;
+			options?: boolean | AddEventListenerOptions;
+		};
+	}>;
```

Comment:
Should you also accept passing the event handler directly without an object, as it is a common case (and before your PR the only possible way) and is easier to read?